File: swift/problem824.swift
import Foundation //to use components() function which split string into string array

class Solution {
    func toGoatLatin(_ S: String) -> String {
        let vowels = Set<Character>(["a","e","i","o","u","A","E","I","O","U"])
        let words = S.components(separatedBy: " ")
        var goat_latin = String() //create output string
        var index = 1
        
        for var word in words {
            if (!vowels.contains(word.first!)) { //word begin with consonant
                let first_consonant = word.removeFirst()
                word.append(first_consonant)
            }
            word.append("ma"+String(repeating: "a", count: index)+" ") //append " " after each word
            index += 1
            goat_latin.append(word)
        }
        
        goat_latin.removeLast(1) //remove redundant " "
        return goat_latin
    }
}

print("I speak goat Latin")
print(Solution().toGoatLatin("I speak goat Latin"))

File: cpp/problem824.cpp
class Solution {
public:
    string toGoatLatin(string S) {
        string res;
        size_t i = 0, j = S.find(' '), count = 0;
        while(true){
            res += _toGoatLatin( S.substr(i,j-i), ++count );
            if( j == S.npos ) break;
            res += " ";
            i = j+1;
            j = S.find( ' ', i );
        }
        return res;
    }
    string _toGoatLatin(const string &S, size_t i){
        switch(tolower(S[0])){
            case 'a': case 'e': case 'i': case 'o': case 'u':
                return S + "ma" + string( i, 'a' );
            default:
                return string( S.begin()+1, S.end() ) + S[0] + "ma" + string(i,'a');
        }
    }

};
